pkgs/development/compilers/dotnet/build-dotnet.nix

```
@@ -121,6 +122,21 @@ stdenv.mkDerivation (finalAttrs: rec {
       "aarch64-darwin" = "osx-arm64";
     };
 
+    updateScript =
+      if type != "sdk" then
+        throw "Only the SDK package can be updated!"
+      else
+      let
+        majorVersion =
+          with lib;
+          concatStringsSep "." (take 2 (splitVersion version));
+      in
+      writeScript "update-dotnet-${majorVersion}" ''
+        pushd pkgs/development/compilers/dotnet
+        exec ${./update.sh} "${majorVersion}"
+        popd
```

`popd` doesn't do anything since this executes in a subshell
```suggestion
        cd pkgs/development/compilers/dotnet
        exec ${./update.sh} "${majorVersion}"
```